File: lib/airspace/network.ts
// lib/airspace/network.ts
// Network view + coordination gap (R7). Reference computation, not a solver. Ported from backend/network.py.
import type { NetworkView } from "./types";

export const COORDINATION_FACTOR = 0.75; // coordinated = 75% of everyone acting alone

const round2 = (n: number) => Math.round(n * 100) / 100;

export function buildNetworkView(selfishTotalUsd: number): NetworkView {
  const coordinated = round2(selfishTotalUsd * COORDINATION_FACTOR);
  return {
    selfishUsd: round2(selfishTotalUsd),
    coordinatedUsd: coordinated,
    gapUsd: round2(selfishTotalUsd - coordinated),
  };
}

File: backend/network.py
from backend.models import NetworkView
COORDINATION_FACTOR = 0.75

def build_network_view(selfish_total_usd):
    coordinated = round(selfish_total_usd * COORDINATION_FACTOR, 2)
    return NetworkView(selfish_usd=round(selfish_total_usd,2), coordinated_usd=coordinated,
                       gap_usd=round(selfish_total_usd - coordinated, 2))
